### PR TITLE
Refactor some led_set_kb instances

### PR DESCRIPTION
**Pull request by zvecr**
## Description

With the aim to help `led_set_kb` removal.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/retro_refit/retro_refit.c [hunk 1]
```diff
@@ -12,36 +12,37 @@ void matrix_init_kb(void) {
...
+    if(res) {
+        if (led_state.caps_lock) {
+            // output low
+            DDRD |= (1<<0);
+            PORTD &= ~(1<<0);
+        } else {
+            // Hi-Z
+            DDRD &= ~(1<<0);
+            PORTD &= ~(1<<0);
+        }
```
Comment (fauxpark) [inline keyboards/handwired/retro_refit/retro_refit.c:26]:
```suggestion
        writePin(D0, !led_state.caps_lock);
```

File: keyboards/handwired/retro_refit/retro_refit.c [hunk 2]
```diff
@@ -12,36 +12,37 @@ void matrix_init_kb(void) {
...
+        if (led_state.num_lock) {
+            // output low
+            DDRD |= (1<<1);
+            PORTD &= ~(1<<1);
+        } else {
+            // Hi-Z
+            DDRD &= ~(1<<1);
+            PORTD &= ~(1<<1);
+        }
```
Comment (fauxpark) [inline keyboards/handwired/retro_refit/retro_refit.c:35]:
```suggestion
        writePin(D1, !led_state.num_lock);
```

File: keyboards/handwired/retro_refit/retro_refit.c [hunk 3]
```diff
@@ -12,36 +12,37 @@ void matrix_init_kb(void) {
...
+        if (led_state.scroll_lock) {
+            // output low
+            DDRC |= (1<<6);
+            PORTC &= ~(1<<6);
+        } else {
+            // Hi-Z
+            DDRC &= ~(1<<6);
+            PORTC &= ~(1<<6);
+        }
```
Comment (fauxpark) [inline keyboards/handwired/retro_refit/retro_refit.c:44]:
```suggestion
        writePin(C6, !led_state.scroll_lock);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/v1x_right/v1x_right.c [hunk 4]
```diff
@@ -62,8 +62,10 @@ void matrix_init_kb(void) {
...
-void led_set_kb(uint8_t usb_led) {
-    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
-    writePin(LED_02, !IS_LED_ON(usb_led, USB_LED_NUM_LOCK));
-    led_set_user(usb_led);
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(LED_02, !led_state.num_lock);
+    }
+    return res;
```
Comment (t-8ch) [inline keyboards/dumbpad/v1x_right/v1x_right.c:70]: This pattern could be replaced by
```c
void led_update_ports(led_t led_state) {
    writePin(LED_02, !led_state.num_lock);
}
```

(same goes for most of the other changes)

**Review by drashna — APPROVED** [review]

**Pull request merged**